Core/include/Acts/TrackFinder/CombinatorialKalmanFilter.hpp

```diff
@@ -1041,8 +1041,10 @@ class CombinatorialKalmanFilter {
       ACTS_VERBOSE("Apply smoothing on " << nStates
                                          << " filtered track states.");
       // Smooth the track states
-      auto smoothRes = m_smoother(state.geoContext, result.fittedStates,
-                                  measurementIndices.front());
+      GlobalBoundSymMatrix globalTrackParamsCov;
```

Comment:
This is never used in the CKF, right?

Reply:
Right. But if we add a non-const reference parameter (for the global covariance) in the function `m_smoother`, we have to pass a parameter when calling it here as well, right? This is a little awkward, but I don't know a better way to handle this.

Reply:
All right, this redundancy has been removed with the parameter as a pointer.

Reply:
This is one solution. Maybe use `std::optional`? Would that fit into the workflow (i.e. with all the dimension checks you perform)?

Reply:
Let's use pointer for the moment since the interface is still to be determined.